docs/contributing/coding-standards/component-options.md

```diff
@@ -4,204 +4,42 @@ Configuration options allow developers to pass additional information into a com
+- [data attributes on the root element](https://frontend.design-system.service.gov.uk/building-configurable-components/#receiving-configuration-from-data-attributes) of the component
 
-```mjs
-import { mergeConfigs, validateConfig, normaliseDataset } from '../../common/configuration.mjs'
-import { ConfigError } from '../../errors/index.mjs'
-import { GOVUKFrontendComponent } from '../../govuk-frontend-component.mjs'
+These are the same sources as those used by GOV.UK Frontend components. Configuration options from these sources are merged into a single configuration object by the `ConfigurableComponent` class. Note that:
 
-export class Accordion extends GOVUKFrontendComponent {
-  constructor($root, config = {}) {
-    super($root)
+- configuration options passed to the constructor will override any defaults
+- data attribute values have highest precedence and will override any configuration set elsewhere
 
-    this.config = mergeConfigs(
-      Accordion.defaults,
-      config,
-      normaliseDataset(Accordion, this.$root.dataset)
-    )
+Your component will then be able to:
 
-    // Check that the configuration provided is valid
-    const errors = validateConfig(Accordion.schema, this.config)
-    if (errors[0]) {
-      throw new ConfigError(formatErrorMessage(CharacterCount, errors[0]))
-    }
-  }
-}
-```
+[access the merged configuration](https://frontend.design-system.service.gov.uk/building-configurable-components/#accessing-the-component-s-configuration) using `this.config`
+use [the`Component` class](https://frontend.design-system.service.gov.uk/building-your-own-javascript-components/#using-the-component-class) features
 
 ## Adding a Nunjucks parameter
 
-Most, but not all, components support adding arbitrary attributes and values through the `attributes` parameter. This method is also more verbose compared to having a dedicated Nunjucks parameter.
-
-Using the previous bit of HTML, update it to make the `data-*` attribute changeable.
+If you're using Nunjucks macros, you can use a dedicated Nunjucks parameter to set data attributes on the root element.
 
 ```nunjucks
 <div
-  class="govuk-accordion"
-  data-module="govuk-accordion"
+  class="my-component"
+  data-module="my-component"
   {%- if params.rememberExpanded %} data-remember-expanded="{{ params.rememberExpanded | escape }}"{% endif %}>
   ...
 </div>
 ```
 
 The above code checks for the existence of the `rememberExpanded` parameter. If the parameter is present it adds the `data-remember-expanded` attribute. It then passes in the developer's defined value, running it through [Nunjucks' native `escape` filter](https://mozilla.github.io/nunjucks/templating.html#escape-aliased-as-e) to make sure that values can't break our HTML.
 
-We can now call the Accordion's Nunjucks macro with our new `rememberExpanded` parameter:
+We can now call the component's Nunjucks macro with our new `rememberExpanded` parameter:
```

Comment:
**issue** The example following this statement still talks about the 'Accordion'. 

If we switch the introduction of the previous example to present the Accordion's template as an example, we might be able to do something similar here:

```suggestion
With the previous code in the Accordion's template, this is how a page may set a specific value for the `rememberExpanded` option:
```